modules/c++/sys/source/OSUnix.cpp

```diff
@@ -44,6 +44,26 @@
 #include "sys/OSUnix.h"
 #include "sys/File.h"
 
+
+namespace
+{
+void linkStat(const std::string& pathname, struct stat& linkStatus)
+{
+    if (lstat(pathname.c_str(), &linkStatus) != 0)
+    {
+        throw except::Exception(Ctxt(strerror(errno)));
+    }
+}
+
+int readlinkOrError(const std::string& pathname, char* buffer)
```

Comment:
Right.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, July 15, 2016 10:39 AM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/coda-oss] Current path (#146)


In modules/c++/sys/source/OSUnix.cpp<https://github.com/mdaus/coda-oss/pull/146#discussion_r70983648>:

> @@ -44,6 +44,26 @@

>  #include "sys/OSUnix.h"

>  #include "sys/File.h"

>

> +

> +namespace

> +{

> +void linkStat(const std::string& pathname, struct stat& linkStatus)

> +{

> +    if (lstat(pathname.c_str(), &linkStatus) != 0)

> +    {

> +        throw except::Exception(Ctxt(strerror(errno)));

> +    }

> +}

> +

> +int readlinkOrError(const std::string& pathname, char* buffer)

Return type should be void not int right?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/coda-oss/pull/146/files/3966d87f89b750e5c5f36301bab1110f8dde2377#r70983648>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYFyyjsaK2gI5vdhdKRo52oo5RLEoks5qV5uSgaJpZM4JLdyF>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.